### ISSUE TITLE
no ladspa pugins

### ISSUE THREAD

**Issue by Reaper10** [hydrogen-music/hydrogen#1309]
Hydrogen version *          : 1.0.2
Operating system + version  :  windows 10
Audio driver + version      : 

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

no ladspa pugins
![Screenshot 2021-06-29 161812](https://user-images.githubusercontent.com/8137561/123868842-dcec6880-d8f5-11eb-8146-7a458185c206.jpg)

**Comment by dfkettle** [comment]
Having the same problem here with v1.1.0-beta1 on Windows 7 (64-bit). If you need someone to help with testing under Windows, let me know.

**Comment by aomahana** [comment]
Confirming, same issue with Windows 10 64 bit, Hydrogen V 1.1.1

**Comment by theGreatWhiteShark** [comment]
> Confirming, same issue with Windows 10 64 bit, Hydrogen V 1.1.1

That's quite odd. LADSPA [should](https://github.com/hydrogen-music/hydrogen/commit/0dfa73ff4afd7fe6a98407c95f6cec5da0ab74c7#diff-e1ab4c3cec24f841948eb867754d4ad0b9b6d11eaa6b5bfbe0357aceb0a49eff) be enabled in both v1.1.1 and v1.1.0-beta1.

Could you try either [Hydrogen-1.1.1-win64.exe](https://ci.appveyor.com/api/buildjobs/ct1wet4x59lmu3ck/artifacts/Hydrogen-1.1.1-win64.exe) or [Hydrogen-1.1.1-win32.exe](https://ci.appveyor.com/api/buildjobs/2xxbl01lb09yb9ox/artifacts/Hydrogen-1.1.1-win32.exe)? I used them recently to debug and check LADSPA support on my local machine. Do they show the same warning?

**Comment by aomahana** [comment]
Hi,

No, there is no warning message.
What windows version of LADSPA did you use to test with?
Can you send me a link so I can re-check?
Reverb would be the most useful FX.

**Comment by Reaper10** [comment]
1..0.2

**Comment by aomahana** [comment]
Hi,

If 1.0.2 is directed at me - I was asking what windows version of LADSPA you are testing with, and where to find it.
I am using the current version of Hydrogen, v1.1.1

**Comment by theGreatWhiteShark** [comment]
> What windows version of LADSPA did you use to test with?

Hydrogen was build using LADSPA SDK 1.17-1 (using the MSYS2 `pacman` package **mingw-w64-i686-ladspa-sdk-1.17-1**). But I'm not sure whether the LADSPA version is that relevant here since it's a quite old plugin format and I do not expect it to have changed much in the recent years.

The big problem I encountered during testing was that I couldn't find a single LADSPA plugin that was working and had to compile a couple of them myself (see #1453). But I'm not a Windows user, so, maybe I just didn't searched the right places.

**Comment by aomahana** [comment]
Thanks for that.

I had not seen the #1453 discussion.
LADSPA seems essentially to be an obsolete unsupported format, especially at 64 bit.
I also had not been able to find any Win 64 LADSPA plugins.

So unless someone creates an alternative plugin host in Hydrogen, or writes new LADSPA Win 64 plugins, it seems there is no usable FX rack in Hydrogen for Windows 64.

Hydrogen is still wonderful, and has been improved greatly with recent fixes.
An FX rack would be great, especially if it hosted VST, but I realise how much work this could be.
And in the meantime, Hydrogen is still awesome ....

**Comment by trebmuh** [comment]
https://www.audacityteam.org/download/legacy-windows/

**Comment by theGreatWhiteShark** [comment]
> or writes new LADSPA Win 64 plugins

There is no need to write new ones or do a major rewrite of the old one. In general, all that is required is to adjust the make and cmake files of existing projects and to bundle the compiled results. But, yes, with the advent of VST and LV2, especially in terms of compressor and EQ handling, plugins improved _a lot_.

Adding support for other plugin types would probably be quite some work. But this alone won't cut it. The Hydrogen Mixer itself would require a huge overhaul since right now only 4 effects placed exclusively in aux channels can be used. Well, it's not on anyone's todo list but the topic is popping up from time to time and it maybe be tackled eventually.

> See here: https://www.fosshub.com/Audacity-old.html?dwl=LADSPA_plugins-win-0.4.15.exe

I tried those as well as the LADSPA plugins shipped with LMMS. But, unfortunately, neither of them work with Hydrogen. (Maybe some missing compiler options preventing them from being loaded via Qt? Don't know...)

**Comment by aomahana** [comment]
Yes, the Audacity 0.4.15 LADSPA files are 32 bit and the .dll install in C: Program Files (x86) / Audacity / Plug-ins.
When you open Hydrogen / Mixer / FX / Edit .... Hydrogen does not see them.
Is there another location they could be installed, or is there a way of selecting the path in Hydrogen?
Or will 32 bit LADSPA .dll just not work with a 64 bit Windows OS?

**Comment by theGreatWhiteShark** [comment]
> Is there another location they could be installed, or is there a way of selecting the path in Hydrogen?

There is the `data/plugins` folder in the `.hydrogen` directory within the user's home folder. When putting it there Hydrogen should be able to find it. You can check by starting it from the console with the `-VDebug` option. If Hydrogen attempts to load the plugins but fails, this should show up in the log.

> Or will 32 bit LADSPA .dll just not work with a 64 bit Windows OS?

This, unfortunately, I do not know. I'm neither into Windows nor computer architecture in general.

**Comment by aomahana** [comment]
I installed the Audacity LADSPA dlls in hydrogen / data/ plugins in both Win 10 64 and Win 7 32.
Neither show up in    hydogen/ mixer/ FX /edit    window.
Looking at debug /user data directory /plugins   I can see the dlls.
The debug log shows : Effects:: getpluginList Error loading the library.

**Comment by theGreatWhiteShark** [comment]
> The debug log shows : Effects:: getpluginList Error loading the library.

Same here. Qt is not able to load these shared libs.

**Comment by cme** [comment]
> Or will 32 bit LADSPA .dll just not work with a 64 bit Windows OS?

Correct, 64 bit processes can't use 32 bit libraries in general. For instance passing a pointer to a buffer at an arbitrary 64-bit address, there would be no way to address this with a 32 bit pointer. 


Theoretically it might be possible to make it work for a specific API, or to use a separate 32-bit host process, but that wouldn't be a trivial task.

**Comment by theGreatWhiteShark** [comment]
Hydrogen is not shipped with any LADSPA plugins and there are currently none available on Windows. Even on Linux there is not much movement in LADSPA development anymore. People tend to use LV2, VST and other more recent and powerful formats. But you could try [LMMS](https://lmms.io/). It seems they made some progress recently to also compile and bundle LADSPA plugins for Windows.
